src/cattrs/gen/_shared.py

```diff
@@ -62,3 +64,28 @@ def handler(v, _, _h=handler):
     except RecursionError:
         # This means we're dealing with a reference cycle, so use late binding.
         return c.structure
+
+
+def get_fields_annotated_by(cls: type, annotation_type: type[T] | T) -> dict[str, T]:
+    type_hints = get_type_hints(cls, include_extras=True)
+    # Support for both AttributeOverride and AttributeOverride()
+    annotation_type_ = annotation_type if isinstance(annotation_type, type) else type(annotation_type)
+
+    # First pass of filtering to get only fields with annotations
+    fields_with_annotations = (
+        (field_name, param_spec.__metadata__)
+        for field_name, param_spec in type_hints.items()
+        if hasattr(param_spec, "__metadata__")
+    )
+
+    # Now that we have fields with ANY annotations, we need to remove unwanted annotations.
+    fields_with_specific_annotation = (
+        (
+            field_name,
+            next((a for a in annotations if isinstance(a, annotation_type_)), None),
```

Comment:
This function will only find annotations that were instantiated, which works fine for the purpose of finding annotations created by `gen.override()`